### ISSUE TITLE
Sending a string longer than exactly 44,350 characters over `wss://` hangs indefinitely in Deno v1.33.0 and greater

### ISSUE THREAD

**Issue by josephrocca** [denoland/deno#20355]
Very simple to replicate:
```js
// main.js
const ws = new WebSocket('wss://bleeding-imagine-scsi-ruled.trycloudflare.com'); // hosting this for ease of replication while this issue is debugged - see instructions below to host your own if needed

ws.addEventListener('open', () => {
  ws.send('a'.repeat(44351));
  console.log('Sent message.');

  ws.addEventListener('message', (event) => {
    if(event.data === 'success') {
      console.log('Got response.');
    }
  });
});
```
```bash
deno upgrade --version 1.33.0 # bug replicates in all later versions too, including latest (1.36.4) and canary
deno run --allow-net main.js
```
I've only tested on Ubuntu 22.04 - for me it hangs at "Sent message." forever, about 80% of the time. If I drop the string length from 44351 to 44350, it works fine 100% of the time, and it also works fine on versions prior to Deno v1.32.5 and earlier. It also works fine if you paste the above code in the browser console (regardless of string length), of course.

If for some reason it works fine for you (e.g. it could be OS-version dependent), increase the string length to a high number like 1,000,000.

I'm hosting the referenced remote websocket server for ease of replication, but [here's the code](https://gist.github.com/josephrocca/4c476a3250988a56016039d65d2cc142) in case you want to do it yourself. Note that the bug isn't related to the server at all - I've replicated it with Node.js/Python/Deno WebSocket server across various runtime versions.

@bartlomieju @littledivy The above code is a much more minimal replication of the same problem I mentioned here:

 * https://github.com/denoland/deno/issues/18735#issuecomment-1701946239

Opening a separate issue because the bug seems distinct (different version cut-off), though it is perhaps related.

**Comment by bartlomieju** [comment]
Thanks for the report 👍 that's very detailed and useful. I assume the problem doesn't happen on `ws://`, only on `wss://`? That would indicate problem in the integration with TLS library on our side.

**Comment by josephrocca** [comment]
@bartlomieju Yep, that's correct - `ws://` seems to work fine.

**Comment by josephrocca** [comment]
@littledivy Apologies for the ping! Wondering if there's any rough ETA on this? I'm completely blocked 🫠

**Comment by bartlomieju** [comment]
@josephrocca we're still debugging this problem. No ETA yet, but we'll aim to fix it before v1.37.

**Comment by josephrocca** [comment]
Hey @bartlomieju, any update on this or new (very) rough ETA?

**Comment by bartlomieju** [comment]
@mmastrac is working on this problem, but I don't know the ETA of the fix. Matt do you have any estimate?

**Comment by jflatow** [comment]
I'm slightly blocked on https://github.com/denoland/deno/issues/18977 and hoping the fixes here may also address that. Can I help with anything @mmastrac, like testing? It looks like yall merged a websocket change into main not long ago, I'm compiling and will try that version, is it expected to work? Shall I try with `mmastrac:tls` instead?

**Comment by mmastrac** [comment]
Early testing on the PR is definitely welcome. I'm currently blocked on my own work upstream in a new TLS library before I can merge this work, but it is possible you may be able to make things work. Backpressure is currently broken on my branch but otherwise should actually work. 

Please let me know if that helps

**Comment by jflatow** [comment]
Unfortunately [this](https://github.com/denoland/deno/pull/20518) alone doesn't fix the #18977 issue 😞

**Comment by mmastrac** [comment]
I just pushed a new commit to that branch that might help with driving the handshake on writes. We may need to get a test case integrated for this.

**Comment by jflatow** [comment]
Ah there's a bunch of debugging statements slowing it down (and had to patch up the rebase), but it seems like its actually working 🎉 I can work on getting a test case properly integrated

**Comment by josephrocca** [comment]
@mmastrac Sorry for the ping again, I was going to ask separately on both #20579 #20518, but I figured I'd just ask about them both here - wondering if there's a very rough ETA on this?

(I tested with `--canary` just in case it was fixed by denoland/rusty_v8#1326  but the issue is still there)

**Comment by mmastrac** [comment]
@josephrocca I believe this might be fixed in the next canary. Would you mind re-testing once the canary for https://github.com/denoland/deno/commit/b75f3b5ca0952db8b50cf417c107f3f14fe582d5 builds?

**Comment by josephrocca** [comment]
Yep, that fixed it - thank you!!

**Comment by caramboleyo** [comment]
i still have the secure websocket bug with 1.38.5
it just appears more random now.
sometimes when i enter the page the socket just does not answer anything
if i then do a ping through it after some time it suddenly works
this is still not production ready

**Comment by caramboleyo** [comment]
funny thing is when it hangs and i send a ping suddenly both responses arrive, the one that was hanging and the pong.
And the hanging messages sent were around 300bytes
the incoming ones about 77 bytes
so it cant be length anymore, just something is holding them back
